src/client/ui/inputs/NumericInput.js

```diff
@@ -1,7 +1,162 @@
 import React from "react";
+import PropTypes from "prop-types";
+
 import styles from "./NumericInput.scss";
-import ReactNumericInput from "react-numeric-input";
 
-export default function NumericInput(props) {
-  return <ReactNumericInput style={false} className={styles.numericInput} {...props} />;
+function round(value) {
+  return Math.round(value * 1000) / 1000;
+}
+
+const partialValue = /[-.0]$/;
+const wholeNumber = /-?[0-9]+$/;
+
+export default class NumericInput extends React.Component {
+  static propTypes = {
```

Comment:
ReactNumeric input also had support for min/max, precision, and format. I think we'll want to add those options eventually. We may also want to add them as options to the component schema?

Reply:
Yup, adding min/max right now -- we definitely need it for things like the metallic and roughness values. Not sure precision needs to be an option right now. Precision should probably be the same for all the numbers we care about in the editor, should probably never be more than 3 digits.
I can't think of a usecase for format right now, but we can add it in the future if necessary.

Reply:
The only thing I was thinking is maybe for adding the degrees symbol when the input isn't focused?